Source/Diagnostics/FieldIO.cpp

```diff
@@ -143,9 +143,16 @@ WriteOpenPMDFields( const std::string& filename,
   auto dataset = openPMD::Dataset(datatype, global_size);
 
   // Create new file and store the time/iteration info
-  auto series = openPMD::Series( filename,
-                                 openPMD::AccessType::CREATE,
-                                 MPI_COMM_WORLD );
+  auto series = [filename](){
+      if( ParallelDescriptor::NProcs() > 1 )
+        return openPMD::Series( filename,
+                                openPMD::AccessType::CREATE,
+                                MPI_COMM_WORLD );
```

Comment:
This works already for the three backends `bp`, `h5`, and `json` (serial).

In the case where we "oversubscribe" the number of MPI ranks over our available blocks, this deadlocks since not all MPI ranks in `MPI_COMM_WORLD` reach this point inside `WarpX::WritePlotFile()`. Is there a better-suited MPI communicator we can use here?

Reply:
cc @WeiqunZhang do you have an idea what's a good communicator to pick here?

Reply:
ok, the "oversubscription" issue is already present and did not change with the latest updates. We can merge this and address this separately https://github.com/openPMD/openPMD-api/pull/554